fv3core/tests/main/test_dycore_call.py

```diff
@@ -0,0 +1,162 @@
+import contextlib
+import os
+import unittest.mock
+from typing import Any, List, Tuple
+
+import f90nml
+
+import fv3core
+import fv3core._config
+import fv3core.initialization.baroclinic as baroclinic_init
+import fv3core.testing
+import pace.util
+from fv3core._config import Namelist
+from fv3core.grid import MetricTerms
+from fv3core.utils.grid import DampingCoefficients, GridData
+from fv3core.utils.null_comm import NullComm
+
+
+DIR = os.path.abspath(os.path.dirname(__file__))
+NAMELIST_FILENAME = os.path.join(DIR, "c12_namelist.nml")
+
+
+@contextlib.contextmanager
+def no_lagrangian_contributions(dynamical_core: fv3core.DynamicalCore):
+    # TODO: lagrangian contributions currently cause an out of bounds iteration
+    # when halo updates are disabled. Fix that bug and remove this decorator.
+    # Probably requires an update to gt4py (currently v36).
```

Comment:
it's doing this because the data has non-monotonic pressure. I am not aware of a gt4py change that fixes this, are you? a potential solution (not for this PR) could be for the null communicator to copy columns from the compute domain to the halo of the same rank so that there's real (garbage) data at all points. This workaround looks like it works though.

Reply:
there were two things, one we don't have absolute k indexing, which we may have at some point (turbulence needs it), and the other was knowing what the last k index is. But it might be possible to have it passed as an external.

Reply:
It can currently be passed as an external, and we don't actually need absolute k indexing because of this, but the current version does not support if statements inside for loops inside if statements, or if statements of any kind inside while loops, and various other combinations that could make this possible.

Reply:
I think Johann added that capability already. You need to know the current absolute k index to be able to compare it to the external max allowed k. Maybe that is also possible now.

Reply:
Johann was helping us fix it :) Apparently he has implemented these capabilities, they just aren't in v36.